### PR TITLE
fix: Corrected expected shape of user input in get Prompt()

### PR DESCRIPTION
**Pull request by cmcadams-newrelic**
## Description

The shape of the users prompt message was incorrect for claude 3.5 
```
 // Prepare the payload for the model.
  const payload = {
    anthropic_version: "bedrock-2023-05-31",
    max_tokens: 1000,
    messages: [
      {
        role: "user",
        content: [{ type: "text", text: prompt }],
      },
    ],
  };
```
https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/javascript_bedrock-runtime_code_examples.html#anthropic_claude

## How to Test

run unit test suite.

### CODE REVIEW

File: lib/llm-events/aws-bedrock/bedrock-command.js [hunk 1]
```diff
@@ -86,9 +86,9 @@ class BedrockCommand {
     } else if (this.isClaude3() === true) {
       result = this.#body?.messages?.reduce((acc, curr) => {
         if (typeof curr?.content === 'string') {
-          acc += curr?.content
-        } else if (Object.keys(curr?.content).length) {
-          acc += curr?.content.text ?? ''
+          return curr?.content
+        } else if (Array.isArray(curr?.content) && Object.keys(curr?.content[0]).length) {
+          return curr?.content[curr?.content.length - 1].text ?? ''
```
Comment (cmcadams-newrelic) [inline lib/llm-events/aws-bedrock/bedrock-command.js:91]: we only care about the last message, which would be the most recent user prompt.

Comment (jsumners-nr) [reply lib/llm-events/aws-bedrock/bedrock-command.js:91]: If that is the case, do we need to retain the `.reduce`?

Comment (cmcadams-newrelic) [reply lib/llm-events/aws-bedrock/bedrock-command.js:91]: Im thinking in the future we might want to use the reduce to filter out messages from `{type: text}` and `{type: image}` at this current point in time. I don't see a benefit with using the `reduce` function.

File: test/unit/llm-events/aws-bedrock/bedrock-command.test.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ const claude = {
 const claude35 = {
   modelId: 'anthropic.claude-3-5-sonnet-20240620-v1:0',
   body: {
-    messages: [{ role: 'user', content: { type: 'text', text: 'who are you' } }]
+    messages: [{ role: 'user', content: [{ type: 'text', text: 'who are you' }] }]
```
Comment (cmcadams-newrelic) [inline test/unit/llm-events/aws-bedrock/bedrock-command.test.js:30]: addresses the bug where the shape of the object was incorrect. content is an array of objects

**Comment by jsumners-nr** [comment]
@cmcadams-newrelic in Slack you mentioned wanting to get this PR ready for review by the end of today. I see that you have marked it ready, but I don't see any new commits. It sounded like you would be making some changes. Is this ready?

File: lib/llm-events/aws-bedrock/bedrock-command.js (hunk 1 above)
Comment (jsumners-nr) [reply lib/llm-events/aws-bedrock/bedrock-command.js:91]: How are other agent teams handling these cases? I haven't seen a spec that specifically calls out this model's operations.

Comment (RyanKadri) [reply lib/llm-events/aws-bedrock/bedrock-command.js:91]: I think for 1, it's just a relatively new API that got introduced after we added Bedrock support. Python also seems to only handle the string case and not the multi-chunk case. I'm not positive but I'd guess that's the same in other agents (who support Bedrock) as well.

For 2, I think Claude is actually getting closer to how we handle Chat Completions in OpenAI [here](https://github.com/newrelic/node-newrelic/blob/57e6be9f4717fde3caada0e3ca3680959180f928/lib/instrumentation/openai.js#L139). For one completion call, we already split out several message events based on the context messages that were passed in

**Review by jsumners-nr — CHANGES_REQUESTED** [review]
Tests are failing.

**Comment by cmcadams-newrelic** [comment]
@jsumners-nr Sorry for the delay. I have the fixed the tests and pushed up all required changes

**Comment by jsumners-nr** [comment]
@cmcadams-newrelic the unit tests are failing. Please rebase against `main` and run `npm run unit`.

**Comment by cmcadams-newrelic** [comment]
@jsumners-nr I have rebased this branch with main as well as corrected the failing test.

File: lib/llm-events/aws-bedrock/bedrock-command.js [hunk 3]
```diff
@@ -84,14 +84,19 @@ class BedrockCommand {
...
+      result = this.#body?.messages
+        ?.reduce((acc, curr) => {
+          if (curr?.role !== 'assistant') {
+            if (typeof curr?.content === 'string') {
+              acc.push(curr?.content)
+            } else {
+              const mappedMsgObj = curr?.content.map((msgObj) => msgObj.text)
+              acc.push(mappedMsgObj)
+            }
+          }
```
Comment (jsumners-nr) [inline lib/llm-events/aws-bedrock/bedrock-command.js:96]:
```suggestion
          if (curr?.role !== 'assistant') {
            return acc
          }
          if (typeof curr?.content === 'string') {
            acc.push(curr?.content)
          } else {
            const mappedMsgObj = curr?.content.map((msgObj) => msgObj.text)
            acc.push(mappedMsgObj)
          }
```

File: test/unit/llm-events/aws-bedrock/bedrock-command.test.js (hunk 2 above)
Comment (jsumners-nr) [inline test/unit/llm-events/aws-bedrock/bedrock-command.test.js:30]:
```suggestion
    messages: [
      { role: 'user', content: [{ type: 'text', text: 'who are' }] },
      { role: 'assistant', content: [{ type: 'text', text: 'researching' }] },
      { role: 'user', content: [{ type: 'text', text: 'you' }] }
    ]
```

File: test/unit/llm-events/aws-bedrock/bedrock-command.test.js [hunk 4]
```diff
@@ -207,7 +207,7 @@ test('claude35 minimal command works', async (t) => {
   assert.equal(cmd.maxTokens, undefined)
   assert.equal(cmd.modelId, claude35.modelId)
   assert.equal(cmd.modelType, 'completion')
-  assert.equal(cmd.prompt, claude35.body.messages[0].content.text)
+  assert.equal(cmd.prompt, claude35.body.messages[0].content[0].text)
```
Comment (jsumners-nr) [inline test/unit/llm-events/aws-bedrock/bedrock-command.test.js:210]:
```suggestion
  assert.equal(cmd.prompt, 'who are you')
```

File: test/unit/llm-events/aws-bedrock/bedrock-command.test.js [hunk 5]
```diff
@@ -221,7 +221,7 @@ test('claude35 complete command works', async (t) => {
   assert.equal(cmd.maxTokens, 25)
   assert.equal(cmd.modelId, payload.modelId)
   assert.equal(cmd.modelType, 'completion')
-  assert.equal(cmd.prompt, payload.body.messages[0].content.text)
+  assert.equal(cmd.prompt, payload.body.messages[0].content[0].text)
```
Comment (jsumners-nr) [inline test/unit/llm-events/aws-bedrock/bedrock-command.test.js:224]:
```suggestion
  assert.equal(cmd.prompt, 'who are you')
```

**Review by jsumners-nr — APPROVED** [review]

**Pull request merged**